### PR TITLE
Getting the infrastructure fot NetFX test runner inplace

### PR DESCRIPTION
**Pull request by tarekgh**
This change will enable running the tests with NetFX.
Note that we still need to go through all tests and fix any issues there like assembly binding issues or the test is written in way that not suitable for the desktop runs

The change also include some fixes in return the src and test sources compile back with NetFx

### CODE REVIEW

**Comment by tarekgh** [comment]
@weshaggard I hope you can review this one before you go off. 

Note that, this change will get the NetFX runner work but we still need to fix all failing tests because of another assembly binding or bad tests for desktop

File: dir.props [hunk 1]
```diff
@@ -276,7 +276,8 @@
     <NETCoreAppTestSharedFrameworkPath>$(NETCoreAppTestRootPath)shared/Microsoft.NETCore.App/$(NETCoreAppTestSharedFxVersion)/</NETCoreAppTestSharedFrameworkPath>
 
     <!-- Constructed shared fx path for testing -->
-    <TestSharedFxDir>$(NETCoreAppTestRootPath)</TestSharedFxDir>
+    <TestSharedFxDir Condition="'$(_bc_TargetGroup)' != 'netfx'">$(NETCoreAppTestRootPath)</TestSharedFxDir>
+    <TestSharedFxDir Condition="'$(_bc_TargetGroup)' == 'netfx'">$(RuntimePath)</TestSharedFxDir>
```
Comment (tarekgh) [inline dir.props]: @weshaggard this is not really needed but it was wrong anyway for NetFX. so fixing it in case we'll use it somewhere later

**Comment by tarekgh** [comment]
test Innerloop OSX Release Build and Test please

File: dir.props (hunk 1 above)
Comment (weshaggard) [inline dir.props]: I wouldn't bother setting it for netfx I would just add a Condition for BinplaceTestSharedFramework=true as it only applies in that case.

File: external/test-runtime/XUnit.Runtime.depproj [hunk 2]
```diff
@@ -32,11 +32,23 @@
...
         <Private>false</Private>
         <NuGetPackageId>$(XUnitRunnerPackageId)</NuGetPackageId>
         <NuGetPackageVersion>$(XUnitPackageVersion)</NuGetPackageVersion>
       </ReferenceCopyLocalPaths>
     </ItemGroup>
-  </Target>
+    
+    <WriteLinesToFile
+      File="$(RuntimePath)\xunit.console.exe.config"
+      Lines="$([System.IO.File]::ReadAllText('$(ToolsDir)\XUnitNetFXConfigFileTemplate.txt').Replace('{RunTimePath}', $(RuntimePath)))"
```
Comment (weshaggard) [inline external/test-runtime/XUnit.Runtime.depproj]: nit: I would change `RunTimePath` to `RuntimePath` here and in your template.

File: Tools-Override/XUnitNetFXConfigFileTemplate.txt [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+   <runtime>  
+      <assemblyBinding xmlns="urn:schemas-microsoft-com:asm.v1"> 
+
+         <dependentAssembly>  
+            <assemblyIdentity name="xunit.execution.dotnet" publicKeyToken="8d05b1bb7a6fdb6c" culture="neutral" />
+            <codeBase version="2.2.0.3300" href="{RunTimePath}xunit.execution.dotnet.dll"/>
+         </dependentAssembly>
+
+         <dependentAssembly>  
+            <assemblyIdentity name="System.IO.FileSystem" publicKeyToken="b03f5f7f11d50a3a" culture="neutral" />
```
Comment (weshaggard) [inline Tools-Override/XUnitNetFXConfigFileTemplate.txt]: How did you determine the set of files needed in this list? We need to figure out a better way to maintain this set.

Comment (tarekgh) [inline Tools-Override/XUnitNetFXConfigFileTemplate.txt]: talked offline with @weshaggard and we are going to automatic generate the list in a future PR

Comment (weshaggard) [inline Tools-Override/XUnitNetFXConfigFileTemplate.txt]: OK please make sure we have an issue tracking this.

**Pull request merged**